### PR TITLE
Underscore in number literal

### LINKED ISSUE #6: Underscore in number literals
Go numbers allow for underscores:

~~~go
package main

func main() {
   n := 1_000
   println(n)
}
~~~

- https://golang.org/ref/spec#Floating-point_literals
- https://golang.org/ref/spec#Integer_literals

however in the above program, the `1_000` is not recognized as a number. I think
the issue is here:

https://github.com/google/vim-ft-go/blob/e9b820bc30467fa151f1ebd63222648ec71e022f/syntax/go.vim#L137-L139
https://github.com/google/vim-ft-go/blob/e9b820bc30467fa151f1ebd63222648ec71e022f/syntax/go.vim#L148-L151

### PR DESCRIPTION
**Pull request by 89z**
Fix #6

### CODE REVIEW

File: syntax/go.vim [hunk 1]
```diff
@@ -135,8 +135,8 @@ syn region      goBlock             start="{" end="}" transparent fold
 syn region      goParen             start='(' end=')' transparent
 
 " Integers
-syn match       goDecimalInt        "\<\d\+\([Ee]\d\+\)\?\>"
-syn match       goHexadecimalInt    "\<0x\x\+\>"
+syn match       goDecimalInt        "\<[0-9_]\+\([Ee]\d\+\)\?\>"
```
Comment (dbarnett) [inline syntax/go.vim:138]: You don't want this to match leading underscores like `_42` though, right? That's an identifier per https://golang.org/ref/spec#Integer_literals. Probably want to change that part of the pattern to `[1-9][0-9_]*`.

**Review by dbarnett — CHANGES_REQUESTED** [review]
Thanks! Suggested one tweak. 👍 apart from that.

**Comment by 89z** [comment]
I tried this:

~~~
\<[1-9][0-9_]*\+\([Ee]\d\+\)\?\>
~~~

but I get:

~~~
E871: (NFA regexp) Can't have a multi follow a multi
~~~

I dont know what the hell that means. I am pretty good with RegEx, but not Vim RegEx.

**Comment by 89z** [comment]
Updated to deal with starting underscore

**Review by dbarnett — APPROVED** [review]
Thanks! Merging...

**Comment by 89z** [comment]
Hey @dbarnett why did you say `1-9`? Now plain `0` doesnt get highlighted

**Comment by dbarnett** [comment]
> Hey @dbarnett why did you say `1-9`? Now plain `0` doesnt get highlighted

Ah, good catch. I was following the structure of the `decimal_lit` spec at https://golang.org/ref/spec#Integer_literals, but I guess forgot to include the part for 0 itself (`"0" | …`). I should really get some tests added…

Thanks for catching and fixing in #8!

**Pull request merged**